jdaviz/core/template_mixin.py

```diff
@@ -1,26 +1,34 @@
-import ipyvuetify as v
+from warnings import warn
+
+from ipyvuetify import VuetifyTemplate
 from glue.core import HubListener
-from traitlets import List
 
 __all__ = ['TemplateMixin']
 
 
-class TemplateMixin(v.VuetifyTemplate, HubListener):
+class TemplateMixin(VuetifyTemplate, HubListener):
     def __new__(cls, *args, **kwargs):
         """
         Overload object creation so that we can inject a reference to the
         ``Hub`` class before components can be initialized. This makes it so
         hub references on plugins can be passed along to components in the
         call to the initialization method.
         """
-        app = kwargs.pop('app', None)
         obj = super().__new__(cls, *args, **kwargs)
-        setattr(obj, '_app', app)
+        obj._app = kwargs.pop('app', None)
 
         return obj
 
     @property
     def app(self):
+        warn(DeprecationWarning('The usage `.app` from a jdaviz application is'
+                                ' deprecated and should be replaced by '
+                                '`.glueapp` to prevent confusion with other '
+                                'meanings of "app"'))
+        return self._app
+
+    @property
+    def glueapp(self):
```

Comment:
This is never true. When [plugins are instantiated](https://github.com/spacetelescope/jdaviz/blob/master/jdaviz/app.py#L853), they are passed `self` as the `app` reference. Therefore, the `TemplateMixin` would be passing `app=SpecViz()`, or `app=CubeViz()` depending on context, of `app=Application()` with no configuration. In any case, it's never passed the glue application instance. The only accessor to the underlying glue app should exist on the JDAViz `Application` class in `app.py`.

Reply:
Aha, that's my misunderstanding.  My main reason for making this change is that I think having two layers of `app` is confusing, so I'm thinking this should have a more specific name than `app` (or alternatively, the `helper` could have it's name changed...).  What do you think is a more explicit term for this?

(Also, can you use `suggest` in Github here to add a short docstring saying what this was intended here?)

Reply:
Currently, there is no accessor for the underlying glue application, so there's nominally only one "app" and that's the current JDAViz application the user is interacting with. I think instead we should add a property to the `app.py` file called `glue_app` or something.

Reply:
I'm still confused here @nmearl: my intent behind this was to not have an attribute with an ambiguous name `app` refer to two different *classes*/APIs. Since`SpecViz` and `Application` definitely have different APIs, I'm trying to separate them out into two different attribute names.  I see your point that the link to the *glue* application should live in the `Application`, but the fact that I was confused about this is illustrating the problem I'm trying to solve... so are you saying I should revert this? If so, what should we say as the docstring of which meaning of "app" the `TemplateMixin` is?